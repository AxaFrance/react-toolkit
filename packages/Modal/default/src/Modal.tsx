import React from 'react';
import ReactModal from 'react-modal';

import ModalCore, { ModalCoreProps } from './ModalCore';
import Header from './Header';
import HeaderBase from './HeaderBase';
import Body from './Body';
import Footer from './Footer';

ReactModal.setAppElement('body');

const Modal = (props: ModalCoreProps) => <ModalCore {...props} />;

Modal.Body = Body;
Modal.Footer = Footer;
Modal.Header = Header;
Modal.HeaderBase = HeaderBase;

export default Modal;
