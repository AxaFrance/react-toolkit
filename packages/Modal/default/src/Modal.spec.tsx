import React from 'react';
import {mount} from 'enzyme';
import Modal from './Modal';

describe('<Modal>', () => {
  it('should render modal', () => {
    const modal = mount(<Modal isOpen onOutsideTap={jest.fn()}>Content</Modal>);
    expect(modal).toMatchSnapshot();
  });
});
