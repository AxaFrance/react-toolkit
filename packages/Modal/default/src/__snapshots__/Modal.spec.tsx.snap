// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal> should render modal 1`] = `
<Modal
  isOpen={true}
  onOutsideTap={[MockFunction]}
>
  <ModalCoreRaw
    className="af-modal"
    isOpen={true}
    onOutsideTap={[MockFunction]}
    title=""
  >
    <Modal
      aria={
        Object {
          "modal": true,
        }
      }
      ariaHideApp={true}
      bodyOpenClassName="ReactModal__Body--open"
      className="af-modal"
      closeTimeoutMS={0}
      contentLabel=""
      isOpen={true}
      onRequestClose={[MockFunction]}
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
    >
      <Portal
        containerInfo={
          <div
            class="ReactModalPortal"
          >
            <div
              class="ReactModal__Overlay ReactModal__Overlay--after-open"
              style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75);"
            >
              <div
                aria-label=""
                aria-modal="true"
                class="ReactModal__Content ReactModal__Content--after-open af-modal"
                role="dialog"
                tabindex="-1"
              >
                <div
                  class="af-modal__dialog"
                >
                  <div
                    class="af-modal__content"
                  >
                    Content
                  </div>
                </div>
              </div>
            </div>
          </div>
        }
      >
        <ModalPortal
          aria={
            Object {
              "modal": true,
            }
          }
          ariaHideApp={true}
          bodyOpenClassName="ReactModal__Body--open"
          className="af-modal"
          closeTimeoutMS={0}
          contentLabel=""
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={true}
          onRequestClose={[MockFunction]}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          role="dialog"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "content": Object {},
              "overlay": Object {},
            }
          }
        >
          <div
            className="ReactModal__Overlay ReactModal__Overlay--after-open"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              }
            }
          >
            <div
              aria-label=""
              aria-modal={true}
              className="ReactModal__Content ReactModal__Content--after-open af-modal"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              role="dialog"
              style={Object {}}
              tabIndex="-1"
            >
              <div
                className="af-modal__dialog"
              >
                <div
                  className="af-modal__content"
                >
                  Content
                </div>
              </div>
            </div>
          </div>
        </ModalPortal>
      </Portal>
    </Modal>
  </ModalCoreRaw>
</Modal>
`;
